core/src/main/java/org/elasticsearch/plugins/InstallPluginCommand.java

```diff
@@ -134,6 +134,26 @@
     private final OptionSpec<Void> batchOption;
     private final OptionSpec<String> arguments;
 
+    public static final Set<PosixFilePermission> dirAndExecutablePerms = new HashSet<>(7);
+    public static final Set<PosixFilePermission> filePerms = new HashSet<>(4);
+
+    static {
+        // Directories and executables get chmod 755
+        dirAndExecutablePerms.add(PosixFilePermission.OWNER_EXECUTE);
+        dirAndExecutablePerms.add(PosixFilePermission.OWNER_READ);
+        dirAndExecutablePerms.add(PosixFilePermission.OWNER_WRITE);
+        dirAndExecutablePerms.add(PosixFilePermission.GROUP_EXECUTE);
+        dirAndExecutablePerms.add(PosixFilePermission.GROUP_READ);
+        dirAndExecutablePerms.add(PosixFilePermission.OTHERS_READ);
+        dirAndExecutablePerms.add(PosixFilePermission.OTHERS_EXECUTE);
+
+        // Files get chmod 644
+        filePerms.add(PosixFilePermission.OWNER_READ);
```

Comment:
Same thing here: can we create a local set, add the permissions to that set, and then make `filePerms` an unmodifiable copy of the local set?